### ISSUE TITLE
Do not have top-level toolkit imports

### ISSUE THREAD

**Issue by lilyminium** [openforcefield/openff-nagl#56]
## Expected behavior ##

There's no real need to have top-level toolkit imports and it's creating an import loop in the toolkit: https://github.com/openforcefield/openff-toolkit/pull/1700#discussion_r1310660172

It should be fairly easy to move these imports to the individual functions.

I believe the guarded `if typing.TYPE_CHECKING` should be ok for type signatures?

**Comment by mattwthompson** [comment]
> I believe the guarded ...

Yup - it's a little annoying to see them in quotation marks but if they're only used for annotations then this is a huge win. (Are there really no "real" imports of toolkit objects?)
